test/pkey_meth_kdf_test.c

```diff
@@ -94,10 +107,23 @@ static int test_kdf_hkdf(void)
         TEST_error("EVP_PKEY_CTX_set1_hkdf_key");
         goto err;
     }
-    if (EVP_PKEY_CTX_add1_hkdf_info(pctx, (const unsigned char *)"label", 5)
+    if (index == 0) {
+        if (EVP_PKEY_CTX_add1_hkdf_info(pctx, (const unsigned char *)"label", 5)
             <= 0) {
-        TEST_error("EVP_PKEY_CTX_set1_hkdf_info");
-        goto err;
+            TEST_error("EVP_PKEY_CTX_set1_hkdf_info");
+            goto err;
+        }
+    } else {
+        if (EVP_PKEY_CTX_add1_hkdf_info(pctx, (const unsigned char *)"lab", 3)
```

Comment:
You will need to check the fips provider version here and skip the index 1 test if an old fips provider is used.

Reply:
Ideally this would have an expected result after the derive, so that you know it added the info correctly..

Reply:
It does on lines 134/135...
```
        const unsigned char expected[sizeof(out)] = {
            0x2a, 0xc4, 0x36, 0x9f, 0x52, 0x59, 0x96, 0xf8, 0xde, 0x13

Reply:
It's calling `add1` twice in this portion of the test, vs the `index == 0` case where it calls `add1` once. The values end up being the same, and the `EVP_PKEY_derive()` is at the end where the expected value is compared.